src/utils/prim-lexing.fs

```diff
@@ -156,14 +156,21 @@ namespace Internal.Utilities.Text.Lexing
             new LexBuffer<'Char>(filler)
               
         // A full type signature is required on this method because it is used at more specific types within its own scope
-        static member FromArray (s: 'Char[]) : LexBuffer<'Char> = 
+        // Important: This method takes ownership of the array
+        static member FromArrayNoCopy (buffer: 'Char[]) : LexBuffer<'Char> = 
             let lexBuffer = new LexBuffer<'Char>(fun _ -> ())
-            let buffer = Array.copy s 
-            lexBuffer.Buffer <- buffer;
-            lexBuffer.BufferMaxScanLength <- buffer.Length;
+            lexBuffer.Buffer <- buffer
+            lexBuffer.BufferMaxScanLength <- buffer.Length
             lexBuffer
 
-        static member FromChars (arr:char[]) = LexBuffer.FromArray arr 
+        // A full type signature is required on this method because it is used at more specific types within its own scope
```

Comment:
I think the comment about full signature can go.

Reply:
Waiting for further confirmation, if I should remove all the same comments on the two other methods as well.

Notice that there are dubious comments here:

https://github.com/Microsoft/visualfsharp/blob/75e435e74b81bb47c4a0ced11be0d7560d2a66ea/src/fsharp/UnicodeLexing.fs#L35-L40

It doesn't look like those comments apply as the stream is totally handled within the method and we just return the LexBuffer object.

Reply:
Yeah those comments look old